tlslite/tlsconnection.py

```diff
@@ -1628,6 +1663,21 @@ def _serverGetClientHello(self, settings, certChain, verifierDB,
                 else:
                     ffGroupIntersect = True
 
+        # Check and save clients heartbeat extension mode
+        heartbeatExt = clientHello.getExtension(ExtensionType.heartbeat)
+        if heartbeatExt:
+            if heartbeatExt.mode == HeartbeatMode.peer_allowed_to_send and \
+                    settings.heartbeatResponseCallback:
```

Comment:
case with "we know how to handle heartbeats, but we will never send them" is missing
(in other words, check for `peer_allowed_to_send` should happen in isolation)